### ISSUE TITLE
Automatically Sync Usage with GCP

### ISSUE THREAD

**Issue by ashwin153** [infracost/infracost#2196]
- This is probably already on the roadmap, but I didn't see an issue filed for it.
- `google_storage_bucket` and `google_bigquery_table` are good starting points because their cost is very usage-dependent.
- Writing usage files by hand is tedious, but cost estimates on GCP are useless without them.

**Comment by alikhajeh1** [comment]
@ashwin153 we're thinking of addressing that issue by digesting the billing data from the cloud vendors, e.g. [this](https://www.infracost.io/docs/infracost_cloud/actual_costs/) is what we did for AWS. What do you think of that approach?

**Comment by alikhajeh1** [comment]
@ashwin153 good point! We've thought about how the analysis can be run in your cloud account. The output usage data (think of it as similar to the usage file) would be sent to your organization in Infracost Cloud automatically, so CLI runs in CI/CD can query/fetch that from Infracost Cloud. That keeps runs fast, and ensures other features such as [Guardrails](https://www.infracost.io/docs/infracost_cloud/guardrails/) and [Policies](https://www.infracost.io/docs/infracost_cloud/cost_policies/) that benefit from usage data work ok too. What do you think of that approach?

Also, note that we're going through our [SOC2 compliance certification](https://infracost.io/security/) and are happy to chat with security/compliance teams to understand their concerns and how we can address them.

**Comment by alikhajeh1** [comment]
@ashwin153 sure, [this](https://github.com/infracost/infracost/blob/master/infracost-usage-example.yml) shows all of the resources/params that would be in that data, notice how it has the resource address (e.g. `aws_rds_cluster.my_dev_cluster`) and the usage values for it.

**Comment by glenngillen** [comment]
We're moving to a contribution model where Issues will be exclusively for work that is ready to be worked on. Discussions and feature requests about potential future capabilities will be over in Discussions. There has been very little community or customer interest in this capability for an extended period, but if it still interests you (or you're now reading this issue because you've been looking for a solution) please create a new thread in Discussions where we can better gather interest from the community.
